### PR TITLE
CUMULUS-3682 -- Update @cumulus/message.Granule to handle missing values in CMR metadata

### PR DESCRIPTION
**Pull request by Jkovarik**
This commit adds logic in convertDateToISOStringSettingNull to

- set '' as an expected null if encountered in any of the granule temporal fields handled by the Granules methods.
- Handle explicit undefined values by returning them rather than attempting to date convert them (the error case for this ticket)

This update should be compatible with api granule write logic in that it's preserving the intent of implicit undefined values while allowing for explicit undefineds that weren't originally part of the expected typing coming from the CMR metadata spec.

**Summary:** Summary of changes

Addresses [CUMULUS-3682](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3682)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: packages/message/src/Granules.ts [hunk 1]
```diff
@@ -133,13 +135,15 @@ function isProcessingTimeInfo(
...
+ * and converting '' to null
  *
  * @param {string} date - Date string, possibly in multiple formats
  * @returns {string} Standardized ISO date string
  */
-const convertDateToISOStringPreservingNull = (date: string | null) => {
-  if (date === null) return null;
+const convertDateToISOStringSettingNull = (date: string | null | undefined) => {
+  if (isNil(date)) return date;
+  if (date === '') return null;
```
Comment (npauzenga) [inline packages/message/src/Granules.ts:146]: I think this is good and is more explicit than something like `if (!date)` but is there a case here where just checking for a falsy value with `!date` falls down?

**Review by npauzenga — APPROVED** [review]
Nice @Jkovarik this looks very reasonable. I added a question about using your explicit checks vs. `!date` but that's mainly to ensure there's not a case here that needs to be explicitly tested. 

Pre-emptively approving because I don't see where this might cause issues.

File: packages/message/src/Granules.ts (hunk 1 above)
Comment (Jkovarik) [reply packages/message/src/Granules.ts:146]: Not really, no, for the set `(date: string | null | undefined) `  L145 handles `null | undefined`, and (!date) would handle the string case, however it felt like in this case it obscured the intent/reduced explicitness in favor of a blanket implicit behavior. 

tl;dr favoring explicit intent over implicit coverage here, it felt more defensive against unintended future changes.
